modules/plugin/jdbc/jdbc-hana/src/test/java/org/geotools/data/hana/HanaTestSetupBase.java

```diff
@@ -46,6 +77,14 @@ public void setFixture(Properties fixture) {
         String database = fixture.getProperty("database");
         String useSsl = fixture.getProperty("use ssl");
 
+        if ((fixture.getProperty("schemabase") != null)
+                && (fixture.getProperty("schema") == null)) {
+            String schemaBase = fixture.getProperty("schemabase");
+            int counter = schemaCounter.getAndIncrement();
+            String schema = schemaBase + "_" + counter;
```

Comment:
Oh interesting, so each test class is using the same connection but working in a different schema, to avoid stepping on each other toes. And the setup class clears all the schemas before starting up the tests, right?

Reply:
That's the basic idea, the details are slightly different:

- The `CISetup` class generates a unique schema name prefix which is used by that CI run (this enables parallel CI runs)
- Each test gets its own unique schema name to avoid interference with other tests
- Each test gets its own connection(s) from the connection pool that is shared in the CI run (there are too many potential issues in sharing the same connection between two or more threads concurrently)
- The `CITeardown` class deletes the schemas created by the tests
- Because a CI run might be aborted (and the teardown is then not executed), the `CISetup` class will additionally delete geotools test schemas that are older than 24 hours to get rid of leftovers.